config/v1/types_infrastructure.go

```diff
@@ -457,6 +462,32 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+type ClusterDNSConfig struct {
+	// APIServerDNSConfig contains DNS configuration for the API Server.
+	// +optional
+	APIServerDNSConfig []DNSConfig `json:"apiServerDNSConfig,omitempty"`
+
+	// InternalAPIServerDNSConfig contains DNS configuration for the Internal API Server.
+	// +optional
+	InternalAPIServerDNSConfig []DNSConfig `json:"internalAPIServerDNSConfig,omitempty"`
+
+	// IngressDNSConfig contains DNS configuration for cluster services.
```

Comment:
What happens when this is omitted? What's the expectation?

Reply:
Same as with API and API-Int but the Ingress controller is responsible for providing the Load Balancer configuration that is used to generate the Ingress DNS config. 

When userConfiguredDNS is Enabled and this config is not present, cluster installation will fail.